upstream-configuration/scripts/ComposerScripts.php

```diff
@@ -97,34 +42,157 @@ public static function preUpdate(Event $event)
 
             putenv('COMPOSER_ROOT_VERSION=dev-main');
         }
+
+        // Apply updates to top-level composer.json
+        static::applyComposerJsonUpdates($event);
     }
 
-   /**
-    * Update the composer.lock file and so on.
-    *
-    * Upstreams should *not* commit the composer.lock file. If a local working
-    * copy
-    */
-    private static function updateLocalDependencies($io, $packages)
+    /**
+     * postUpdate
+     *
+     * After "composer update" runs, we have the opportunity to do additional
+     * fixups to the project files.
+     *
+     * @param Composer\Script\Event $event
+     *   The Event object passed in from Composer
+     */
+    public static function postUpdate(Event $event)
     {
-        if (!file_exists('composer.lock')) {
+        // for future use
+    }
+
+    /**
+     * Apply composer.json Updates
+     *
+     * During the Composer pre-update hook, check to see if there are any
+     * updates that need to be made to the composer.json file. We cannot simply
+     * change the composer.json file in the upstream, because doing so would
+     * result in many merge conflicts.
+     */
+    public static function applyComposerJsonUpdates(Event $event)
+    {
+        $io = $event->getIO();
+
+        $composerJsonContents = file_get_contents("composer.json");
+        $composerJson = json_decode($composerJsonContents, true);
+        $originalComposerJson = $composerJson;
+
+        // Check to see if the platform PHP version (which should be major.minor.patch)
+        // is the same as the Pantheon PHP version (which is only major.minor).
+        // If they do not match, force an update to the platform PHP version. If they
+        // have the same major.minor version, then
+        $platformPhpVersion = static::getCurrentPlatformPhp($event);
+        $pantheonPhpVersion = static::getPantheonPhpVersion($event);
+        $updatedPlatformPhpVersion = static::bestPhpPatchVersion($pantheonPhpVersion);
+        if ((substr($platformPhpVersion, 0, strlen($pantheonPhpVersion)) != $pantheonPhpVersion) && !empty($updatedPlatformPhpVersion)) {
+            $io->write("<info>Setting platform.php from '$platformPhpVersion' to '$updatedPlatformPhpVersion' to conform to pantheon php version.</info>");
+            $composerJson['config']['platform']['php'] = $updatedPlatformPhpVersion;
+        }
+
+        // Remove our upstream convenience scripts, if the user has not removed them.
+        if (isset($composerJson['scripts']['upstream-require'])) {
```

Comment:
This will be running these changes every time someone runs `composer update` (which includes `composer require`). Are you ok with this?

In DCM, we added a new post-update-hook and made sure to run this code only if that hook was absent: https://github.com/pantheon-systems/drupal-composer-managed/blob/default/upstream-configuration/scripts/ComposerScripts.php#L108

Reply:
Makes sense. Updated to include the same.